src/train.py

```diff
@@ -1,20 +1,66 @@
 
 
-def get_loader(dataset, batch_size=64, split=(4, 1)):
+def get_loader(dataset, batch_size=8, split=(4, 1)):
     val_q = len(dataset) // sum(split) * split[1]
 
     val_indices = np.random.choice(list(range(len(dataset))), val_q)
     train_indices = list(set(range(len(dataset))).difference(val_indices))
-    sampler = SubsetRandomSampler(train_indices)
-    return DataLoader(dataset, sampler=sampler, batch_size=batch_size)
+    train_sampler = SubsetRandomSampler(train_indices)
+    vaild_sampler = SubsetRandomSampler(val_indices)
+    return DataLoader(dataset, sampler=train_sampler, batch_size=batch_size),\
+        DataLoader(dataset, sampler=vaild_sampler, batch_size=batch_size)
 
 
 if __name__ == '__main__':
-    dataset = CasiaSurfDataset()
-    data_loader = get_loader(dataset)
-    for images, labels in enumerate(data_loader):
-        print(images, labels)
-        raise
+    argparser = argparse.ArgumentParser()
+    argparser.add_argument('--protocol', type=int, required=True)
+    argparser.add_argument('--epochs', type=int, default=10)
+    argparser.add_argument('--checkpoint', type=str)
+    argparser.add_argument('--batch_size', type=int, required=True)
+    argparser.add_argument('--eval_every', type=int, default=1)
+    args = argparser.parse_args()
+    dataset = CasiaSurfDataset(args.protocol)
+    device = torch.device('cuda' if torch.cuda.is_available() else 'cpu')
+    train_loader, val_loader = get_loader(dataset)
+    model = models.mobilenet_v2(
+        num_classes=2) if not args.checkpoint else torch.load(args.checkpoint)
+    model = model.to(device)
+    optimizer = optim.Adam(model.parameters())
+    loss_fn = nn.CrossEntropyLoss()
+    print(model)
+    for epoch in range(args.epochs):
+        model.train()
+
+        for i, batch in enumerate(train_loader):
+            images, labels = batch
+            labels = torch.LongTensor(labels)
+            images, labels = images.to(device), labels.to(device)
+            outputs = model(images)
+            loss = loss_fn(outputs, labels)
+            print(
+                f'Epoch: {epoch + 1}/{args.epochs}\tBatch: {i + 1}/{len(train_loader)}\tLoss: {loss.item()}')
+            loss.backward()
+            optimizer.step()
+        torch.save(model.state_dict,
+                   f'checkpoints/mobilenet_v2_protocol{args.protocol}({epoch}).pt')
+
+        if epoch % args.eval_every == 1:
+            model.eval()
+            print("Evaluating...")
+            val_loss = []
+            with torch.no_grad():
+                for i, batch in enumerate(val_loader):
+                    images, labels = batch
+                    labels = torch.LongTensor(labels)
+                    images, labels = images.to(device), labels.to(device)
+                    outputs = model(images)
+                    loss = loss_fn(outputs, labels)
+                    val_loss.append(loss.item())
+                print(f"\t\t\tValidation loss: {np.mean(val_loss)}")
```

Comment:
I think in validation stage we also interested in other metrics? (e.g. accuracy?)